--- installer-1.8.1-original.sh 2020-03-05 22:14:22.743119075 +0300
+++ installer-1.8.1.sh  2020-03-05 22:22:54.544550923 +0300
@@ -31,7 +31,10 @@
 # packages and package metadata; no personal persistent data is stored there.)
 
 RELEASE="1.9.2"
-
+# If METEOR_VERSION is set, we want to install that instead.
+if [ "${METEOR_VERSION}" != "latest" ] ; then
+  RELEASE="${METEOR_VERSION:-$RELEASE}"
+fi
 
 # Now, on to the actual installer!

